pueue_lib/src/network/socket/unix.rs

```diff
@@ -142,8 +142,35 @@ pub async fn get_listener(settings: &Shared) -> Result<GenericListener, Error> {
             })?;
         }
 
-        let unix_listener = UnixListener::bind(&socket_path)
-            .map_err(|err| Error::IoPathError(socket_path, "creating unix socket", err))?;
+        // The various nix platforms handle socket permissions in different
+        // ways, but generally prevent the socket's permissions from being
+        // changed once it is being listened on.
+        let socket = UnixSocket::new_stream()
+            .map_err(|err| Error::IoError("creating unix socket".to_string(), err))?;
+        socket.bind(&socket_path).map_err(|err| {
+            Error::IoPathError(socket_path.clone(), "binding unix socket to path", err)
+        })?;
+
+        if let Some(mode) = settings.unix_socket_permissions {
+            nix::sys::stat::fchmodat(
+                None,
+                &socket_path,
+                nix::sys::stat::Mode::from_bits_truncate(mode),
+                nix::sys::stat::FchmodatFlags::FollowSymlink,
+            )
+            .map_err(|err| {
+                Error::IoPathError(
+                    socket_path.clone(),
+                    "setting socket permissions",
+                    std::io::Error::from_raw_os_error(err as i32),
+                )
+            })?;
```

Comment:
I figured out how to do this without nix:

```
        if let Some(mode) = settings.unix_socket_permissions {
            set_permissions(&socket_path, Permissions::from_mode(mode)).map_err(|err| {
                Error::IoPathError(
                    socket_path.clone(),
                    "setting permissions on unix socket",
                    err,
                )
            })?;
        }
```

This uses just the std lib and works at least for me :)
Could you test whether this also works for your environment?

If so, please remove the nix dependency :)

Reply:
Sure, let me see if it works for me.

Reply:
Can confirm that also works, so I've removed the `nix` dependency 🥳  PR is updated